contentcuration/contentcuration/viewsets/sync/endpoint.py

```diff
@@ -123,21 +124,38 @@ def return_changes(self, request, channel_revs):
         return {"changes": changes, "errors": errors, "successes": successes}
 
     def return_tasks(self, request, channel_revs):
+        customTaskObjectsids = CustomTaskResult.objects.filter(channel_id__in=channel_revs.keys()).values_list("task_id", flat=True)
+
         tasks = TaskResult.objects.filter(
-            channel_id__in=channel_revs.keys(),
+            task_id__in=customTaskObjectsids,
             status__in=[states.STARTED, states.FAILURE],
```

Comment:
For this query, I think flipping it so that we start with the `CustomTaskResult` and have an `Exists` subquery where clause that queries `TaskResult` for the statuses and matching task (aka `task_id=OuterRef('task_id')`), and then we wouldn't need to re-query `CustomTaskResult` on line 137 below.

Also need to `exclude` the task names in the `Exists` subquery
```
task_name__in=[apply_channel_changes_task.name, apply_user_changes_task.name]
```